src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -57,26 +62,20 @@
     public StaticHostProvider(Collection<InetSocketAddress> serverAddresses)
             throws UnknownHostException {
         for (InetSocketAddress address : serverAddresses) {
-            InetAddress ia = address.getAddress();
-            InetAddress resolvedAddresses[] = InetAddress.getAllByName((ia!=null) ? ia.getHostAddress():
-                address.getHostName());
+            InetAddress resolvedAddresses[];
+            try {
+                Method m = InetSocketAddress.class.getDeclaredMethod("getHostString");
+                m.setAccessible(true);
+                resolvedAddresses = InetAddress.getAllByName((String) m.invoke(address));
+            } catch (IllegalAccessException e) {
```

Comment:
Maybe we can catch multiple exceptions in a single shot - such as `catch(IllegalAccessException | NoSuchMethodException | InvocationTargetException e)` to save some typings, given the exception handling logic is exact the same.